keras_cv/layers/preprocessing/solarization.py

```diff
@@ -49,16 +58,28 @@ class Solarization(tf.keras.layers.Layer):
             or [height, width, channels].
     """
 
-    def __init__(self, threshold=None):
+    def __init__(
+        self,
+        addition=0.0,
+        threshold=0.0,
+        value_range=(0, 255),
+    ):
         super().__init__()
+        self.addition = addition
         self.threshold = threshold
+        self.value_range = value_range
 
-    def call(self, images):
-        images = tf.clip_by_value(images, clip_value_min=0, clip_value_max=255)
-        if self.threshold is None:
-            return self._solarize(images)
-        else:
-            return self._solarize_above_threshold(images)
+    def augment_image(self, image, transformation=None):
+        image = preprocessing.transform_value_range(
+            image, original_range=self.value_range, target_range=(0, 255)
+        )
+        result = image + self.addition
+        result = tf.clip_by_value(result, 0, 255)
+        result = self._solarize_above_threshold(result)
+        result = preprocessing.transform_value_range(
+            result, original_range=(0, 255), target_range=self.value_range
+        )
+        return result
 
     def _solarize(self, images):
```

Comment:
Should we remove this method since this is one-liner and only called by _solarize_above_threshold?

Reply:
Good point